xds/internal/xdsclient/authority.go

```diff
@@ -448,6 +448,10 @@ func (a *authority) close() {
 	a.resourcesMu.Lock()
 	a.closed = true
 	a.resourcesMu.Unlock()
+
+	for _, cleanup := range a.serverCfg.Cleanups {
+		cleanup()
+	}
```

Comment:
I think we still need the cleanups to called only when the xdsClient is closing and not when the authority is closing. An authority would be closed after the last watch to it is cancelled (and followed by some time in the idle authorities cache), and it would need to be revived when a watch is registered that points to this authority.

Since the bootstrap config is only parsed once, and therefore only once do we build the credentials, we need to ensure that only one we call its cleanup function.

Reply:
OK, from what I understand:

1. parsing the config instantiates the bundle for all authorities that specify a server, and the top-level xds_servers
2. clientimpl is closed when the last grpc client that uses xDS is closed
3. at that point we must do the converse of 1., which is close all created bundles.

I attempted to implement that in the latest commit.